s3/s3.go

```diff
@@ -1074,11 +1074,14 @@ func (s3 *S3) prepare(req *request) error {
 			return err
 		}
 
-		signpathPatiallyEscaped := partiallyEscapedPath(req.path)
+		signpathPartiallyEscaped := partiallyEscapedPath(req.path)
+		if strings.IndexAny(s3.Region.S3BucketEndpoint, "${bucket}") >= 0 {
```

Comment:
I'm wondering if this should just be `s3.Region.S3BucketEndpoint != ""` and expect that any bucket endpoint is going to be the subdomain style.